File: android/src/main/java/com/mrousavy/camera/frameprocessor/VisionImageProcessor.java
package com.mrousavy.camera.frameprocessor;

import androidx.camera.core.ImageProxy;

import com.google.mlkit.common.MlKitException;
import com.mrousavy.camera.helpers.FrameMetadata;
import com.mrousavy.camera.helpers.GraphicOverlay;

import java.nio.ByteBuffer;

public interface VisionImageProcessor {

    /** Processes ByteBuffer image data, e.g. used for Camera1 live preview case. */
    void processByteBuffer(
            ByteBuffer data, FrameMetadata frameMetadata, GraphicOverlay graphicOverlay)
            throws MlKitException;

    /** Processes ImageProxy image data, e.g. used for CameraX live preview case. */
    void processImageProxy(ImageProxy image, GraphicOverlay graphicOverlay) throws MlKitException;

    /** Stops the underlying machine learning model and release resources. */
    void stop();
}

File: android/src/main/java/com/mrousavy/camera/helpers/VisionImageProcessor.kt
package com.mrousavy.camera.helpers

import androidx.camera.core.ImageProxy
import com.google.mlkit.common.MlKitException
import java.nio.ByteBuffer

interface VisionImageProcessor {
  /** Processes ByteBuffer image data, e.g. used for Camera1 live preview case.  */
  @Throws(MlKitException::class)
  fun processByteBuffer(
    data: ByteBuffer?, frameMetadata: FrameMetadata?, graphicOverlay: GraphicOverlay?
  )

  /** Processes ImageProxy image data, e.g. used for CameraX live preview case.  */
  @Throws(MlKitException::class)
  fun processImageProxy(image: ImageProxy?, graphicOverlay: GraphicOverlay?)

  /** Stops the underlying machine learning model and release resources.  */
  fun stop()
}
